driver/drivers.go

```diff
@@ -145,6 +146,12 @@ func (d *Drivers) Delete(taskName string) error {
 		return errors.New("task name cannot be empty")
 	}
 
+	driver, ok := d.drivers[taskName]
+
+	if ok {
+		driver.DeleteTask(context.Background())
```

Comment:
Another step we could add here is to remove all the template IDs associated with this task from the driverTemplates map. This would be more of a memory cleanup.

https://github.com/hashicorp/consul-terraform-sync/blob/main/driver/drivers.go#L51-L53

Reply:
I like this idea!